src/App.js

```diff
@@ -1,16 +1,105 @@
 import React from 'react';
+import { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
+import ColorChoice from './components/ColorChoice';
 import chatMessages from './data/messages.json';
 
+const local = chatMessages[0].sender;
+const remote = chatMessages[1].sender;
+// check getting names of each sender in list of messages
+// console.log(local, remote);
+
+// sender needs a position attached to is to assign a color later
+const chatMessagesLocalRemote = chatMessages.map((entry) => {
+  if (entry.sender === local) {
+    return { ...entry, position: 'local' };
+  } else {
+    return { ...entry, position: 'remote' };
+  }
+});
+
+// check position in list of entries
+// console.log(chatMessagesLocalRemote);
+
 const App = () => {
+  const [entries, setEntries] = useState(chatMessagesLocalRemote);
+
+  const likeEntry = (id) => {
+    setEntries((entries) => {
+      return entries.map((entry) => {
+        if (entry.id === id) {
+          console.log('id:', id, entry.liked);
+          return { ...entry, liked: !entry.liked };
+        } else {
+          return entry;
+        }
+      });
+    });
+  };
```

Comment:
Great idea to map each entry in the array. I love the way you did all of this! If you wanted to cut down a bit on the nesting, you could have set a variable called newEntries to the return and then used that variable within the setEntries function but overall if this works for you, then I have no problem with it!